### ISSUE TITLE
Rule Change: (no-magic-numbers)

### ISSUE THREAD

**Issue by Naggertooth** [eslint/eslint#15170]
### What rule do you want to change?

no-magic-numbers

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
'no-magic-numbers': ['error', {
  'ignore': ['/delay(.*)/'],
  // OR
  'ignoreCallee': ['delay']
}]

await delay(process.env.NODE_ENV === PROD ? 1 : 5000);
```

### What does the rule currently do for this code?

error

### What will the rule do after it's changed?

ignoring callee which assuming a magic numbers with no pain

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

https://github.com/eslint/eslint/blob/main/lib/rules/no-magic-numbers.js
